src/Sylius/Bundle/ApiBundle/CommandHandler/Checkout/AddressOrderHandler.php

```diff
@@ -76,8 +83,27 @@ public function __invoke(AddressOrder $addressOrder): OrderInterface
             $order->setCustomer($this->provideCustomerByEmail($addressOrder->email));
         }
 
-        $order->setBillingAddress($addressOrder->billingAddress);
-        $order->setShippingAddress($addressOrder->shippingAddress ?? clone $addressOrder->billingAddress);
+        /** @var AddressInterface $newBillingAddress */
+        $newBillingAddress = $addressOrder->billingAddress;
+        /** @var AddressInterface|null $newShippingAddress */
+        $newShippingAddress = $addressOrder->shippingAddress ?? clone $newBillingAddress;
```

Comment:
Both assigns are pointless to be done here, and should be done in `if`'s below not before them IMHO.

Reply:
I believe the reason is that we need to type-hint them with `Core/AddressInterface`... Why these properties are type-hinted with `Addressing/AddressInterface` in the command?